modules/nf-core/custom/matrixfilter/templates/matrixfilter.R

```diff
@@ -160,55 +161,67 @@ if ((opt\$sample_file != '') && ( ! is.null(opt\$grouping_variable))){
 # value even if the proportion is to be used
 
 if (is.null(opt\$minimum_samples_not_na)) {
-    opt\$minimum_samples_not_na <- ncol(abundance_matrix) * opt\$minimum_proportion_not_na
+  opt\$minimum_samples_not_na <- ncol(abundance_matrix) * opt\$minimum_proportion_not_na
 }
 
-# Define the tests
+# We can assume that in most cases no variant filter is required 
+# if no variant filter is set we will assume that the n filter threshold is equivlent to the row lenght
+if (is.null(opt\$most_variant_features)) {
+  opt\$most_variant_features <- nrow(abundance_matrix)
+}
 
+# Define the tests
 tests <- list(
-    'abundance' = function(x) sum(x >= opt\$minimum_abundance, na.rm = T) >= opt\$minimum_samples, # check if rows have sufficiently high abundance
-    'na' = function(x) !any(is.na(x)) || sum(!is.na(x)) >= opt\$minimum_samples_not_na  # check if enough values in row are not NA
+  'abundance' = function(x) sum(x >= opt\$minimum_abundance, na.rm = TRUE) >= opt\$minimum_samples, # check if rows have sufficiently high abundance
+  'na' = function(x) !any(is.na(x)) || sum(!is.na(x)) >= opt\$minimum_samples_not_na  # check if enough values in row are not NA
 )
 
-# Apply the functions row-wise on the abundance_matrix and store the result in a boolean matrix
+# Function to identify rows that are among the top n most variant
+most_variant_test <- function(matrix_data) {
+  # Determine the indices of the top variant rows based on variance
+  top_indices <- order(-apply(matrix_data, 1, var, na.rm = TRUE))[1:opt\$most_variant_features]
+  # Return a boolean vector indicating if each row is among the top variant ones
+  1:nrow(matrix_data) %in% top_indices
+}
 
+# Apply the 'abundance' and 'na' tests row-wise to the abundance_matrix
 boolean_matrix <- t(apply(abundance_matrix, 1, function(row) {
-    sapply(tests, function(f) f(row))
+  sapply(tests, function(f) f(row))
 }))
 
-# We will retain features passing all tests
-
-keep <- apply(boolean_matrix, 1, all)
+# Apply the 'most_variant_test' function to identify the most variant rows and add the result to the boolean matrix
+most_variant_vector <- most_variant_test(abundance_matrix)
```

Comment:
I'd rather you wrapped this in the conditional above. The result should be the same as what you do here, but if I don't configure variance filtering I'd rather this code didn't run at all.

Reply:
Thanks. I've streamlined it to a single conditional block and fixed the whitespace formatting.